### DISCUSSION TITLE
Documentation isn't clear on this ....

### DISCUSSION THREAD

**Discussion by stephenhauck** [Azure/azure-mobile-apps#472] [Q&A] [answered]
The documentation isn't clear on this ...

In the code you have the following to retrieve the ToDoItems ...
![image](https://user-images.githubusercontent.com/12648793/189736277-a4ab5f97-9e00-40a2-b702-2aa7b3f75572.png)

I wrote this and it works but ... where is the data "filtered" .. on the server or on the client after all items are retrieved ?
![image](https://user-images.githubusercontent.com/12648793/189736318-f01a3f62-750d-4e56-8a45-72eb87b99ccb.png)

**Comment by adrianhall** [comment, marked as answer]
Anything before the ToListAsync() is run on the server.  Anything after the ToListAsync() is run on the client.

**Reply by stephenhauck** [reply]
OK, thanks ... that's VERY interesting.
